build.sbt

```diff
@@ -183,40 +150,38 @@ lazy val testkit = project
 lazy val unit = project
   .in(file("scalafix-tests/unit"))
   .settings(
-    noPublish,
+    noPublishAndNoMima,
     // Change working directory to match when `fork := false`.
     Test / baseDirectory := (ThisBuild / baseDirectory).value,
     javaOptions := Nil,
     testFrameworks += new TestFramework("munit.Framework"),
     buildInfoPackage := "scalafix.tests",
     buildInfoObject := "BuildInfo",
-    libraryDependencies ++= testsDeps,
     libraryDependencies ++= List(
       jgit,
       scalatest.withRevision(
         "3.2.0"
       ), // make sure testkit clients can use recent 3.x versions
-      "org.scalameta" %% "testkit" % scalametaV
-    ),
+      scalametaTeskit
+    ) ++ testsDeps,
     Compile / compile / compileInputs := {
       (Compile / compile / compileInputs)
         .dependsOn(
           testsInput / Compile / compile,
-          testsOutput / Compile / compile
+          testsOutput / Compile / compile,
+          testsShared / Compile / compile
         )
         .value
     },
     Test / resourceGenerators += Def.task {
       // copy-pasted code from ScalafixTestkitPlugin to avoid cyclic dependencies between build and sbt-scalafix.
```

Comment:
I don't understand why we use java.properties intead of BuildInfo here ?

Reply:
Indeed, maybe here it would be possible to use `BuildInfo` to generate a `TestkitProperties` instance and load it from the test runner. But as a regular testkit user (rule author, using sbt or not), being forced to generate code does not seem very nice. So I guess the rationale for using java properies here is that supporting java properties is a key feature, and we might as well eat our own dog food and test it here?

Reply:
ok